### PR TITLE
Secp256k1

### PR DESCRIPTION
**Pull request by semenov-vladyslav**
# Description of change

Add Secp256k1 ECDSA support: key generation (`GenerateKey`), SLIP-10 key derivation (`Slip10Generate`, `Slip10Derive`), signing (`Secp256k1EcdsaSign`), EVM address generation (`GetEvmAddress`).

## Links to any relevant issues

Be sure to reference any related issues by adding `fixes issue #`.

## Type of change

Choose a type of change, and delete any options that are not relevant.

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] Enhancement (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Fix

## How the change has been tested

[Main test case](https://github.com/iotaledger/stronghold.rs/blob/secp256k1/client/src/tests/procedure_tests.rs#L292).

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: engine/Cargo.toml [hunk 1]
```diff
@@ -37,7 +37,7 @@ serde = { version = "1.0", features = [ "derive" ] }
   default-features = false
 
   [dependencies.iota-crypto]
-  version = "0.18.0"
+  version = "0.19.0"
```
Comment (thibault-martinez) [inline engine/Cargo.toml:40]: Stronghold also uses covector (.change files)

Comment (semenov-vladyslav) [reply engine/Cargo.toml:40]: Right

Comment (semenov-vladyslav) [reply engine/Cargo.toml:40]: Added changes.

Comment (thibault-martinez) [reply engine/Cargo.toml:40]: I was in a hurry and thought this was bumping the stronghold version 🤦🏻 
But a .change was needed anyway...

File: engine/runtime/Cargo.toml [hunk 2]
```diff
@@ -24,7 +24,7 @@ serde = { version = "1.0", features = [ "derive" ] }
 random = { version = "0.8.4", package = "rand" }
 dirs = { version = "4.0.0" }
 thiserror = { version = "1.0" }
-iota-crypto = { version = "0.18.0", default-features = false, features = [ "blake2b" ] }
+iota-crypto = { version = "0.19.0", default-features = false, features = [ "blake2b" ] }
```
Comment (thibault-martinez) [inline engine/runtime/Cargo.toml:27]: The .change also need to refer to this crate to release it

File: engine/Cargo.toml (hunk 1 above)
Comment (thibault-martinez) [inline engine/Cargo.toml:40]: Same

File: engine/runtime/Cargo.toml (hunk 2 above)
Comment (thibault-martinez) [reply engine/runtime/Cargo.toml:27]: See here https://github.com/iotaledger/stronghold.rs/pull/472/files#diff-67ac304b6523bc3009f9bb05898cb21ce739a60fbb6f5e13393db61f4655bfe2L2-L4

Comment (semenov-vladyslav) [reply engine/runtime/Cargo.toml:27]: Done.

Comment (thibault-martinez) [reply engine/runtime/Cargo.toml:27]: You added engine but not runtime?

Comment (semenov-vladyslav) [reply engine/runtime/Cargo.toml:27]: Yeah, my bad. What should it be named ["runtime"](https://github.com/iotaledger/stronghold.rs/blob/dev/.changes/readme.md?plain=1#L11) or ["stronghold-runtime"](https://github.com/iotaledger/stronghold.rs/pull/472/files#diff-67ac304b6523bc3009f9bb05898cb21ce739a60fbb6f5e13393db61f4655bfe2L2-L4)?

Comment (semenov-vladyslav) [reply engine/runtime/Cargo.toml:27]: Put "stronghold-runtime".

Comment (thibault-martinez) [reply engine/runtime/Cargo.toml:27]: Yeah that's correct

File: client/src/procedures/primitives.rs [hunk 3]
```diff
@@ -480,15 +489,19 @@ impl DeriveSecret<1> for Slip10Derive {
     fn derive(self, guards: [Buffer<u8>; 1]) -> Result<Products<ChainCode>, FatalProcedureError> {
         let dk = match self.input {
             Slip10DeriveInput::Key(_) => {
-                slip10::Key::try_from(&*guards[0].borrow()).and_then(|parent| parent.derive(&self.chain))
-            }
-            Slip10DeriveInput::Seed(_) => {
-                slip10::Seed::from_bytes(&guards[0].borrow()).derive(slip10::Curve::Ed25519, &self.chain)
+                let r = &*guards[0].borrow();
+                dbg!(r.len());
```
Comment (PhilippGackstatter) [inline client/src/procedures/primitives.rs]: Nit: Is the `dbg` intended or leftover?

File: client/src/procedures/primitives.rs [hunk 4]
```diff
@@ -519,19 +532,33 @@ fn x25519_secret_key(raw: Ref<u8>) -> Result<x25519::SecretKey, crypto::Error> {
...
+    Ok(ed25519::SecretKey::from_bytes(&bs))
+}
+
+fn secp256k1_ecdsa_secret_key(raw: Ref<u8>) -> Result<secp256k1_ecdsa::SecretKey, crypto::Error> {
+    let raw_slice: &[u8] = &raw;
+    if raw_slice.len() < secp256k1_ecdsa::SecretKey::LENGTH {
+        let e = crypto::Error::BufferSize {
+            has: raw_slice.len(),
+            needs: secp256k1_ecdsa::SecretKey::LENGTH,
+            name: "data buffer",
```
Comment (PhilippGackstatter) [inline client/src/procedures/primitives.rs]:
```suggestion
            name: "secp256k1 data buffer",
```

Nit: Consider providing a bit more info in the error for debugging.

File: client/src/procedures/primitives.rs (hunk 3 above)
Comment (semenov-vladyslav) [reply client/src/procedures/primitives.rs]: Intended leftover. Fixed.

File: client/src/procedures/primitives.rs (hunk 4 above)
Comment (semenov-vladyslav) [reply client/src/procedures/primitives.rs]: Done

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**